src/ekat/ekat_parameter_list.hpp

```diff
@@ -113,6 +117,19 @@ inline void ParameterList::set (const std::string& name, const T& value) {
   }
 }
 
+template<typename T>
+inline bool ParameterList::isType (const std::string& name) {
+  // Try to get variable with type T, return true if successful, false if an error is thrown
+  try {
+    auto& dummy = any_cast<T>(m_params.at(name));
+    return true;
+  }
+  catch (std::exception& e) {
```

Comment:
I believe some compilers might complain that `e` is defined but not used. Also, you can catch by _const_ reference. So in the end, something like `catch (const std::exception&)`.